PROVENANCE.md

```diff
@@ -0,0 +1,28 @@
+## Verifying SDK build provenance with the SLSA framework
+
+LaunchDarkly uses the [SLSA framework](https://slsa.dev/spec/v1.0/about) (Supply-chain Levels for Software Artifacts) to help developers make their supply chain more secure by ensuring the authenticity and build integrity of our published SDK packages.
+
+As part of [SLSA requirements for level 3 compliance](https://slsa.dev/spec/v1.0/requirements), LaunchDarkly publishes provenance about our SDK package builds using [GitHub's generic SLSA3 provenance generator](https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md#generation-of-slsa3-provenance-for-arbitrary-projects) for distribution alongside our packages. These attestations are available for download from the GitHub release page for the release version under Assets > `multiple-provenance.intoto.jsonl`.
+
+To verify SLSA provenance attestations, we recommend using [slsa-verifier](https://github.com/slsa-framework/slsa-verifier). Example usage for verifying SDK packages is included below:
+
+```
+# Download packages from PyPi
+$ pip download --only-binary=:all: launchdarkly-server-sdk
+
+# Download provenance from Github release
+$ curl --location -O \
+  https://github.com/launchdarkly/python-server-sdk/releases/download/VERSION/multiple.intoto.jsonl
+
+# Run slsa-verifier to verify provenance against package artifacts 
+$ slsa-verifier verify-artifact \
+--provenance-path multiple-provenance.intoto.jsonl \
+--source-uri github.com/launchdarkly/launchdarkly-server-sdk \
+launchdarkly_server_sdk-VERSION-py3-none-any.whl
+
+TBD OUTPUT
```

Comment:
TBD pending test output - @keelerm84 what's the best way to test out the updates to the release workflow?

Reply:
So there isn't a good way to test this without actually publishing something.

You can run the manual-publish with the dry run option (as I've mentioned in another comment), but actually going through the whole process isn't something that is easily done.

You can use the internal release please testing repo and just modify this code to not actually publish anything as well. I've done that a lot during initial dev.

Reply:
Leaving the example output empty for now until we're able to publish something (so it'll come in a separate commit after).

Let's coordinate on doing a manual-publish dry run next week to at least test most of the functionality here, and we can figure out a path for a full test after, though it sounds like the only way to make sure this works all the way through is to do this on an actual release. At the very least, the provenance generation happens last and won't fail the rest of the release steps